tests/shouldwork/Cores/Xilinx/XpmCdcGray.hs

```diff
@@ -0,0 +1,41 @@
+module XpmCdcGray where
+
+import Clash.Explicit.Prelude
+import Data.Proxy
+
+import XpmCdcGrayTypes (D3, D5)
+
+import qualified XpmCdcGrayTypes as Types
+
+-- | This 'topEntity' exists to make @clash-testsuite@ happy. Without it cannot
+-- find the test benches.
+topEntity :: Unsigned 1
+topEntity = 0
+
+tb0 = done
+ where
+  --                          src dst  width  stages  samples
+  done =             Types.tb @D3 @D5 @16     @4      @100     Proxy Proxy SNat expected
+  expected = $(Types.expected @D3 @D5 @16     @4      @100     Proxy Proxy SNat SNat SNat)
+{-# ANN tb0 (TestBench 'topEntity) #-}
+
+tb1 = done
+ where
+  --                          src dst  width  stages  samples
+  done =             Types.tb @D5 @D3 @16     @4      @100     Proxy Proxy SNat expected
+  expected = $(Types.expected @D5 @D3 @16     @4      @100     Proxy Proxy SNat SNat SNat)
+{-# ANN tb1 (TestBench 'topEntity) #-}
+
+tb2 = done
+ where
+  --                          src dst  width  stages  samples
+  done =             Types.tb @D3 @D5 @16     @10     @100     Proxy Proxy SNat expected
+  expected = $(Types.expected @D3 @D5 @16     @10     @100     Proxy Proxy SNat SNat SNat)
+{-# ANN tb2 (TestBench 'topEntity) #-}
+
+tb3 = done
+ where
+  --                          src dst  width  stages  samples
+  done =             Types.tb @D3 @D5 @16     @2      @100     Proxy Proxy SNat expected
+  expected = $(Types.expected @D3 @D5 @16     @2      @100     Proxy Proxy SNat SNat SNat)
+{-# ANN tb3 (TestBench 'topEntity) #-}
```

Comment:
Is there a reason why `XpmCdcGrayTypes` also defines a `D11` but it's not tested here? I would expect there to be a test that uses that domain or that it's removed if these tests cover all interesting behaviour already

Reply:
Hmm, well I guess we'd want to test domains being exactly twice at fast and _more_ than twice as fast, so I'll add D10 too.